packages/native/types/base.d.ts

```diff
@@ -0,0 +1,227 @@
+// Definitions by: Pat Sissons <https://github.com/patsissons>
+// TypeScript Version: 3.4
+
+import {
+  ComponentPropsWithoutRef,
+  ComponentType,
+  NamedExoticComponent
+} from 'react'
+import * as RN from 'react-native'
+
+type ReactNative = typeof RN
+
+export type ReactNativeStyle = RN.ViewStyle | RN.TextStyle | RN.ImageStyle
```

Comment:
I noticed the old `ReturnType<typeof RN.StyleSheet.flatten>` was actually producing an incorrect result (not sure why) of just `ViewStyle`. So I'm just manually defining the union instead now.